homeassistant/components/growatt_server/sensor.py

```diff
@@ -358,7 +358,24 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up growatt server from yaml."""
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry, add_entities):
+    """Set up growatt server from Config Flow."""
+    hass.async_add_executor_job(_setup_entity, hass, config_entry.data, add_entities)
```

Comment:
`async_add_entities` needs to run in the event loop so we can't schedule all of this on the executor. Please split the server setup from the entity addition.

Reply:
Could you expand on this? What do you mean with the server setup?
And what is the preferred alternative to calling the executor for this?

Reply:
The server api login and api calls (everything that does I/O) needs to run in the executor thread pool. The entity creation and addition should run in the event loop without side effects.

Reply:
I've now implemented the executor successfully